Cython/Utility/TypeConversion.c

```diff
@@ -83,7 +83,28 @@ static CYTHON_INLINE PyObject* __Pyx_PyUnicode_FromString(const char*);
 #define __Pyx_PyUnicode_FromOrdinal(o)       PyUnicode_FromOrdinal((int)o)
 #define __Pyx_PyUnicode_AsUnicode            PyUnicode_AsUnicode
 
-#define __Pyx_NewRef(obj) (Py_INCREF(obj), obj)
+#if CYTHON_COMPILING_IN_CPYTHON && PY_VERSION_HEX >= 0x030a0000
+static CYTHON_INLINE PyObject *__Pyx_NewRef(PyObject *obj) {
+    return Py_NewRef(obj);
+}
+#else
+static CYTHON_INLINE PyObject *__Pyx_NewRef(PyObject *obj) {
+    Py_INCREF(obj);
+    return obj;
+}
+#endif
+
+#if CYTHON_COMPILING_IN_CPYTHON && PY_VERSION_HEX >= 0x030a0000
+static CYTHON_INLINE PyObject *__Pyx_XNewRef(PyObject *obj) {
+    return Py_XNewRef(obj);
+}
+#else
+static CYTHON_INLINE PyObject *__Pyx_XNewRef(PyObject *obj) {
+    Py_XINCREF(obj);
+    return obj;
+}
+#endif
+
```

Comment:
It's always better to limit conditional code differences to the a minimum to make them easy to compare and read. Also, this looks like a case where we could allow other Python implementations to hit the newer code path.
```suggestion
static CYTHON_INLINE PyObject *__Pyx_NewRef(PyObject *obj) {
  #if CYTHON_COMPILING_IN_CPYTHON && PY_VERSION_HEX >= 0x030a0000  ||  defined(Py_NewRef)
    return Py_NewRef(obj);
  #else
    Py_INCREF(obj);
    return obj;
  #endif
}

static CYTHON_INLINE PyObject *__Pyx_XNewRef(PyObject *obj) {
  #if CYTHON_COMPILING_IN_CPYTHON && PY_VERSION_HEX >= 0x030a0000  ||  defined(Py_XNewRef)
    return Py_XNewRef(obj);
  #else
    Py_XINCREF(obj);
    return obj;
  #endif
}

```